### DISCUSSION TITLE
Invalid `prisma.user.update()` invocation when it's kind of valid... (nexus, prisma)

### DISCUSSION THREAD

**Discussion by jjjuk** [prisma/prisma#2496] [Q&A] [answered]
Can't understand what's going wrong here:

```
else return await ctx.prisma.user.update({
          data: {
            favorites: [...user.favorites, exercise] //////////here
          },
          where: {
            id: userId
          }          
        })
```

prisma throws an error:

```
Invalid `prisma.user.update()` invocation:

{
  data: {
    favorites: [
      'doda'
    ]
  },
  where: {
    id: 1
  }
}

Unknown arg `0` in data.favorites.0 for type UserUpdatefavoritesInput. Available args:

type UserUpdatefavoritesInput {
  set?: List<String> ////////why list?
}
```
And if we look at schema.prisma there is **favorites** array of strings.

```
model User {
  email            String?  @unique
  favorites        String[]    /////////here
  gym              String?
  id               Int      @default(autoincrement()) @id
  imgurl           String?
  name             String?
  password         String?
  resetToken       String? @unique @map (name: "resettoken")
  resetTokenExpiry String?         @map (name: "resettokenexpiry")

  @@map (name: "users")
}
```

**Comment by ryands17** [comment, marked as answer]
Hey @jjjuk :wave: 
It should be in the following way:
```ts
await ctx.prisma.user.update({
  data: {
    favorites: { set: [...user.favorites, exercise] }
  },
  where: {
    id: userId
  }         
})
```

**Reply by kumarsahil01** [reply]
My problem just solved removing the await from starting 
`await ctx.prisma.user.update({
  data: {
    favorites: { set: [...user.favorites, exercise] }
  },
  where: {
    id: userId
  }         
})`

**Comment by adetorodev** [comment]
@ryands17 
I tried that in my code, but did not work
I am facing the same error:
```
await prisma.user.update({
      data: { verified: true },
      where: { id: parseInt(userId) },
      
    });
```

Here is the model:
```
model User {
  id             Int       @id @default(autoincrement())
  firstName      String?
  lastName       String?
  email          String   @unique
  password       String
  address        String?
  verified       Boolean?
  phoneNumber    String?
  profilePicture String?
  createdAt      DateTime  @default(now())
  updatedAt      DateTime  @updatedAt
}
```

`
Error updating user status: PrismaClientValidationError: 
Invalid `prisma.user.update()` invocation in
/home/dev/marketplace-api/src/controllers/auth/register.ts:66:23

  63 export const updateUserStatus = async (userId: string): Promise<void> => {
  64   try {
  65     const parsedUserId = parseInt(userId, 10);
→ 66     await prisma.user.update({
           where: {
         +   id: Int
           },
           data: {
             verified: true
           }
         })

Argument `id` is missing.
`

**Reply by F-N-LL** [reply]
Hello @adetorodev have you solved this issue? I got the same problem on my code, user.service.ts looks like this in mine:

```
@Injectable()
export class UserService {
  constructor(private prisma: PrismaService) {}

  async editUser(userId: number, dto: EditUserDto) {
    const user = await this.prisma.user.update({
      where: {
        id: userId,
      },
      data: {
        ...dto,
      },
    });

    delete user.hash;

    return user;
  }
}
```

thanks a lot.

**Comment by ludralph** [comment]
Hi @jjjuk,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
